addon/orm/schema.js

```diff
@@ -202,6 +218,25 @@ export default class Schema {
     return this._hydrate(record, dasherize(type));
   }
 
+  modelClassFor(modelName) {
+    return this._registry[camelize(modelName)].class.prototype;
+  }
+
+  addDependentAssociation(association, modelName) {
+    this._dependentAssociations[modelName] = this._dependentAssociations[modelName] || [];
+    this._dependentAssociations[modelName].push(association);
+  }
+
+  dependentAssociationsFor(modelName) {
+    return this._dependentAssociations[modelName];
+  }
+
+  associationsFor(modelName) {
+    let modelClass = this.modelClassFor(modelName);
+
+    return Object.assign({}, modelClass.belongsToAssociations, modelClass.hasManyAssociations);
```

Comment:
This breaks tests in phantomjs 2.1.1 since `Object.assign()` is not supported yet. I guess lodash assign should be used.